src/types/llmService/LLMService.ts

```diff
@@ -4,6 +4,21 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
+
+/**
+ * Interface representing a service for calling a Large Language Model (LLM).
+ */
 export interface LLMServiceInterface {
-  callLLM(engineeredPrompt: string, promptId?: string): Promise<string>;
+  /**
+   * Calls the LLM with the provided engineered prompt, prompt ID, and input token limit.
+   * @param engineeredPrompt - The prompt that has been engineered for the LLM.
+   * @param promptId - The ID of the prompt (optional).
+   * @param outputTokenLimit - The limit on the number of output tokens (optional).
+   * @returns A promise that resolves to the LLM's response as a string.
+   */
+  callLLM(
+    engineeredPrompt: string,
+    promptId?: string,
+    outputTokenLimit?: number
```

Comment:
👍 How many params do we allow before we make this a object parameter?

Reply:
I tried overloading the signatures, but A4D did not like the structure. I'd say, 4